cogs/msg.py

```diff
@@ -157,9 +160,21 @@ async def caption(self, ctx, shorthand, *, caption):
         safe_caption = quote(safe_caption, safe="")
         quoted_url = quote(shorthand_url, safe="")
 
-        text = f"https://api.memegen.link/images/custom/_/{safe_caption}.gif?background={quoted_url}"
+        file_url = f"https://api.memegen.link/images/custom/_/{safe_caption}.gif?background={quoted_url}"
 
-        await ctx.send(text)
+        async with ctx.typing():
+            # Prevent 403 Forbidden errors
+            req = request.Request(file_url,
+                                  headers={
+                                      "User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:109.0) Gecko/20100101 Firefox/111.0"
+                                  })
+
+            # Download a maximum size of 32 MB as it's an in-memory buffer
+            with request.urlopen(req, timeout=30) as r:
```

Comment:
This sounds like it might block the bot for 30 seconds. Do we have any async HTTP library that we can use instead?

Reply:
Ah true, didn't think of that, thank you :)
I switched it over to aiohttp, but that didn't really support a maximum read size